### ISSUE TITLE
Failing to parse e2e.log for v1.26 PRs

### ISSUE THREAD

**Issue by BobyMCbobs** [cncf-infra/verify-conformance#39]
when parsing the e2e.log, it appears to not be correctly parsing those for Kubernetes v1.26, but remains fine for older versions

- https://github.com/cncf/k8s-conformance/pull/2313#issuecomment-1345648687
- https://github.com/cncf/k8s-conformance/pull/2312#issuecomment-1344527510

There may have been a change in the format in sonobuoy once again.

**Comment by BobyMCbobs** [comment]
For supporting versions <v1.26, Sonobuoy provides a line like this
```
{"msg":"PASSED [sig-cli] Kubectl client Update Demo should create and stop a replication controller [Conformance]","completed":1,"skipped":23,"failed":0}
```
for each test in e2e.log

It appears that in e2e.log files <=v1.26, the line looks like this instead

````
[It] should update pod when spec was updated and update strategy is RollingUpdate [Conformance]
```

**Comment by BobyMCbobs** [comment]
With this change, it may be required to drop support for checking on what tests match between the _e2e.log_ and _junit_01.xml_

**Comment by Riaankl** [comment]
> With this change, it may be required to drop support for checking on what tests match between the _e2e.log_ and _junit_01.xml_

Checking if test results are valid is core to the cncf-ci bot's work.

**Comment by BobyMCbobs** [comment]
Checking operations that the bot is currently doing
- e2e.log tests pass
- all required tests are present in e2e.log
- junit_01.xml tests pass and are not skipped
- all required tests are present in junit_01.xml
- the tests count of e2e.log and junit_01.xml both matches

**Comment by BobyMCbobs** [comment]
TLDR; an important file's formatting has been changed (again).

**Comment by hh** [comment]
> TLDR; an important file's formatting has been changed (again).
Let's loop in the sonobuoy folks. For now let's focus on junit_01.xml only and not verify the log file contains the same info.

**Comment by BobyMCbobs** [comment]
> Let's loop in the sonobuoy folks. 

Conversation in the sonobuoy channel
https://kubernetes.slack.com/archives/C6L3G051C/p1670809082212069

> For now let's focus on junit_01.xml only and not verify the log file contains the same info.
> We may need to hold Conformance Submissions for a bit as we figure out a way to update the format or create a workaround.

Not having access to determine which tests have failed in the e2e.log severally dulls the submission validation, removing 
> - e2e.log tests pass
> - all required tests are present in e2e.log
> - the tests count of e2e.log and junit_01.xml both matches

**Comment by hh** [comment]
From https://onsi.github.io/ginkgo/MIGRATING_TO_V2#generating-machine-readable-reports

> Ginkgo V2 introduces a new JSON format that faithfully captures all available information about a Ginkgo test suite. JSON reports can be generated via ginkgo --json-report=out.json. The resulting JSON file encodes an array of types.Report. Each entry in that array lists detailed information about the test suite and includes a list of types.SpecReport that captures detailed information about each spec. These types are documented here.

**Comment by hh** [comment]
The switch to Ginkgo v2 was back in July,  (Migrate Ginkgo from v1 to v2 [#109111](https://github.com/kubernetes/kubernetes/pull/109111))  which was before 1.25 was cut, so it's maybe it's a follow up change somewhere - From @heyste

**Comment by hh** [comment]
Would it make sense for sonobouy to add an additional `--json-report=e2e.json` to the args it provides to gingko ?

We could document that work around until it’s part of the image and update our validation to use ‘e2e.json’ if it exists.

**Comment by hh** [comment]
Hopefully  kubernetes/kubernetes#114416 is a simple enough fix to get upstream and consumed for submissions.

**Comment by BobyMCbobs** [comment]
> The JSON file from [kubernetes/kubernetes#114416](https://github.com/kubernetes/kubernetes/pull/114416) is in a format specific to Ginkgo. If you want to know which tests ran and which failed, use the `junit_01.xml` file. It's content now follows the (vague) JUnit spec a bit more closely. For the specific type definitions used by Ginkgo, see https://github.com/onsi/ginkgo/blob/master/reporters/junit_report.go
> 
> Regarding the JSON output on stdout: that got removed in [kubernetes/kubernetes@3e560e8](https://github.com/kubernetes/kubernetes/commit/3e560e8341c47a5b9d383ef7af5a476bddacea5c) because it had multiple problems:
> 
>     * The ginkgo v2 CLI redirects stdout, so that output wouldn't be visible without `--output-interceptor-mode=none`.
> 
>     * The stats are per worker, so it wasn't clear what that output is supposed to mean when running multiple workers in parallel.
> 
>     * The extra JSON output was undesirable noise during normal E2E test runs.
> 
> 
> Can someone explain what information you need that cannot be obtained from the `junit_01.xml` file?

Thank you for your response, @pohly.

This ticket has evolved through my understanding of the issue.
It's just the _e2e.log_ file which has had unexpected output changes

After each test run, there's a JSON object kinda like
```
    Message   string `json:"msg"`
    Total     int    `json:"total"`
    Completed int    `json:"completed"`
    Skipped   int    `json:"skipped"`
    Failed    int    `json:"failed"`
```
which is currently consumed in the conformance bot

**Comment by BobyMCbobs** [comment]
Without the _e2e.log_ containing these structures, it means that the bot will not be able to report which individual tests have failed.

**Comment by pohly** [comment]
Why can't the bot use `junit_01.xml` for this?

**Comment by BobyMCbobs** [comment]
> Why can't the bot use `junit_01.xml` for this?

My understanding is the that _junit_01.xml_ only contains the list of test which will run (including which are skipped) and not whether the tests have completed or failed. is this correct @pohly

**Comment by pohly** [comment]
No. `junit_01.xml` is a representation of all test that ran, failed, or were skipped. It's the complete test suite result that we use in Kubernetes to determine the result of a test suite invocation.

**Comment by BobyMCbobs** [comment]
> No. `junit_01.xml` is a representation of all test that ran, failed, or were skipped. It's the complete test suite result that we use in Kubernetes to determine the result of a test suite invocation.

@pohly, thank you for the correction, this is helpful! I'll see what I can do

**Comment by BobyMCbobs** [comment]
The format for _junit_01.xml_ has changed since v1.25.

The old format, I had to specify types for (https://github.com/cncf-infra/verify-conformance/blob/main/pkg/suite/suite.go#L113), for parsing; see here
https://github.com/cncf/k8s-conformance/raw/3752c68b19d87692a2931ef0127b9b532eccdea5/v1.24/volcengine-vestack/junit_01.xml (from https://github.com/cncf/k8s-conformance/pull/2305/files) for v1.24

then in v1.25 the format is different and it's using the Sonobuoy provided type (https://github.com/cncf-infra/verify-conformance/blob/main/pkg/suite/suite.go#L20)
https://github.com/cncf/k8s-conformance/raw/226bf4b91dc2d2959c2bdbea7d6208516ce3369c/v1.25/digitalocean/junit_01.xml (from https://github.com/cncf/k8s-conformance/pull/2302/files) for v1.25

This is why we've been using e2e.log files for this information historically.

**Comment by hh** [comment]
In Summary: Both the junit_01.xml and e2e.log file formats have changed in this release:
- e2e.log no longer contains the json structs with per test passfail info
- junit_01.xml previously did not contain passfail info for each test and now does

Prior to this release we were unable to use `junit_01.xml` as it didn’t have pass/fail, so we used e2e.log. Now that junit_01.xml has pass/fail we will need to update the conformance-bot to where the information has migrated.

**Comment by hh** [comment]
@BobyMCbobs Let's go ahead and update the logic for `>1.26` to only examine the `unit_01.xml`.

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs Let's go ahead and update the logic for `>1.26` to only examine the `unit_01.xml`.

Yes, that's the plan @hh, see https://github.com/cncf-infra/verify-conformance/issues/40

**Comment by BobyMCbobs** [comment]
Fixed by #41
